src/vsphere_cpi/lib/cloud/vsphere/cloud.rb

```diff
@@ -676,12 +676,20 @@ def generate_network_env(devices, networks, dvs_index)
       network_env
     end
 
-    def generate_disk_env(system_disk, ephemeral_disk)
-      {
+    def generate_disk_env(system_disk, ephemeral_disk, vm_config)
+      if vm_config.vmx_options['disk.enableUUID'] == "1"
```

Comment:
Not sure that I love the idea of looking inside of the `vmx_options` hash.

Reply:
vSphere does not normally allocate stable UUIDs on the backing disk of a volume unless enableUUID is turned on.   This approach (and/or going directly into the VIM API extraConfig as I do in the case of persistent disks) is the only consistent way of noting that this attribute is set on the VM -- all Kubo/CFCR BOSH worker VMs have a VM extension to set this as Kubernetes vSphere Cloud Provder requires it so it can manage its PVs properly.    

We could alternatively introduce a new first-class cloud property on the VM type that would reflect the underlying VMX option, as an alternative.

Reply:
@svrc-pivotal Yeah that summary would be very helpful. I don't want anyone in the future to remove this code so the context here is very important.